File: ArraysAndStrings/1.5/Python/one_away.py
def one_away(str1, str2):
    if abs(len(str1)-len(str2))>1:
        return False
    count1 = count2 = 0
    diff = 0
    while count1 < len(str1) and count2 < len(str2):
        if diff > 1:
            return False
        if str1[count1] == str2[count2]:
            count1 += 1
            count2 += 1
        else:
            if str1[count1+1] == str2[count2+1]:
                count1 += 1
                count2 += 1
                diff += 1
            elif str1[count1+1] == str2[count2]:
                count1 += 1
                diff += 1
            elif str1[count1] == str2[count2+1]:
                count2 += 1
                diff += 1
            else:
                return False
    return True

if __name__=="__main__":
    str1 = input("Introduce first string\n")
    str2 = input("Introduce second string\n")
    is_one = one_away(str1, str2)
    if is_one:
        print("True")
    else:
        print("False")

File: ArraysAndStrings/1.5/Golang/one_away.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"math"
)

func one_away(word1, word2 string) bool {
	if math.Abs(float64(len(word1)-len(word2))) > 1 {
		return false
	}
	count1 := 0
	count2 := 0
	diff := 0

	for count1<len(word1) && count2<len(word2) {
		if diff>1 {
			return false
		}
		if word1[count1] == word2[count2] {
			count1++
			count2++
		} else {
			diff++
			if word1[count1+1] == word2[count2+1] {
				count1++
				count2++
			} else if word1[count1+1] == word2[count2] {
				count1++
			} else if word1[count1] == word2[count2+1] {
				count2++
			} else {
				return false
			}
		}
	}
	if diff > 1 {
		return false
	} 
	return true
}

func main(){
	reader := bufio.NewReader(os.Stdin)
	word, _ := reader.ReadString('\n')
	word2, _ := reader.ReadString('\n')

	one := one_away(word, word2)

	if(one){
		fmt.Println("They are one away")
	} else {
		fmt.Println("They are not one away")
	}
}